althea_kernel_interface/src/dns.rs

```diff
@@ -0,0 +1,34 @@
+use super::KernelInterface;
+
+use std::fs::File;
+use std::io::Error;
+use std::io::Read;
+use std::net::IpAddr;
+
+impl KernelInterface {
+    /// Get's a list of IP addresses of nameservers from /etc/resolv.conf, may be v6, v4 or both
+    /// generally ignores malformed lines but produces IO errors
+    pub fn get_resolv_servers(&self) -> Result<Vec<IpAddr>, Error> {
+        let mut f = File::open("/etc/resolv.conf")?;
+        let mut contents = String::new();
+        f.read_to_string(&mut contents)?;
+
+        let mut res = Vec::new();
+        for line in contents.lines() {
```

Comment:
Can you know for a fact there isn't gonna be comments in the file before the nameserver entries? I think a regex might fit here better.

Reply:
I've never seen nameserver #comment ip but I guess the format actually allows it. 

The issue is that we would want a pretty complex ipv4 or v6 regex.

Reply:
It's good, just got confused, sorry!